Add connector release notes for 2201.10.0 (#9215)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md

- Introduced support for parsing XML with record types with default values as the expected type, using the default values where required (i.e., if a value corresponding to the record field is not present in the XML value).
- Introduced the option to choose between semantic and syntactic equality of XML elements and attributes.

#### `salesforce` package

- Added support for Salesforce listener with Streaming API for real-time data synchronization and event-driven workflows.
- Added support for base types with `salesforce.types` submodule.
- Reorganized salesforce clients for better usability.

> ### New connectors

Author: @niveathika IMO we can remove this sub section and include the same as the notes under the particular connectors. We've followed the same release notes format in U9 as well https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.9.0#revamped-connectors.

WDYT?

Reviewer: I am -1 to call these as revamped connectors, as these functionalities were not available previously in ballerina.

Author: @niveathika This structure was proposed in Update 9 to capture all the changes related to the connector revamping effort(where these new connector were also a part of the same effort). But I understand your concern on having `Revamped Connectors` as the title for all the changes. In that case we can rename it as `Connector updates` and keep all the new/revamped connector release under that single section. wdyt?

Reviewer: +1 for that

- Added parameter passing support for XSLT transformations.

### Connector updates

New versions of the following connectors have been released with major updates, as part of the Ballerina connector revamp initiative.
All listed connectors have been released under new major versions, featuring significant API and functionality changes, along with improved documentation and examples.

#### `asb` package

- Introduced support for listener-service-based async message consumption functionality.

#### `dayforce` package

- Introduced as a new connector with support for the Dayforce REST API v1.

#### `discord` package

- Introduced as a new connector with support for the Discord REST API v10.

#### `salesforce` package

> - Added support for Salesforce listener with Streaming API for real-time data synchronization and event-driven workflows.

Reviewer 2: ```suggestion
- Introduced the Salesforce listener with support for the Streaming API for real-time data synchronization and event-driven workflows.
```

> - Added support for base types with `salesforce.types` submodule.

Reviewer 2: ```suggestion
- Added support for base types with the `salesforce.types` submodule.
```

> - Reorganized salesforce clients for better usability.

Reviewer 2: ```suggestion
- Reorganized Salesforce clients for better usability.
```

#### `sap` package

> - Added support for invoking any OData endpoints in S/4HANA.

Reviewer 2: ```suggestion
- Added support for invoking any OData endpoint in S/4HANA.
```

- Added support for automatic CSRF token authentication for enhanced security and usability.

#### `sap.jco` package

- Enabled invocation of RFC modules with improved type binding.
- Added support for sending and receiving iDocs.

#### `sap.s4hana.sales` packages

- Provided a high-level interface for APIs related to Sales and Distribution (SD) in S/4HANA.
- Provided operations for sales order management, delivery processing, and billing.

#### `slack` package

> - Replaced existing remote function-based APIs with resource function-based APIs.

Reviewer 2: ```suggestion
- Replaced existing remote method-based APIs with resource method-based APIs.
```

> - Replaced existing remote function-based APIs with resource function-based APIs.

Reviewer 3: We usually use code blocks for keywords, right?

```suggestion
- Replaced existing `remote` method-based APIs with `resource` method-based APIs.
```

Author: Here I want to highlight the concepts of `remote methods` and `resource methods` in Ballerina (and not the qualifiers), so I'm not sure if highlighting only the keywords makes much sense in this case. @MaryamZi thoughts?

Reviewer 2: Yeah, in this scenario I think we can skip too, since they make sense without backticks too.

- Added support for the latest Slack REST API.

#### `stripe` package

- Replaced existing remote function-based APIs with resource function-based APIs.
- Added support for the latest Stripe v1 REST API.

> #### `twitter(X)` package

Author: Yes the module name is still Twitter. So +1 to keep it as is
